app/resources/service.py

```diff
@@ -170,3 +163,34 @@ def part_owner(data: dict, user: str) -> dict:
 def check_part_owner(data: dict, microservice: str) -> dict:
     # all these requests are trusted
     return {"ok": game_content.part_owner(data["device_uuid"], data["user_uuid"])}
+
+
+@m.microservice_endpoint(path=["hardware", "scale"])
+def hardware_scale(data: dict, mircoservice: str) -> dict:
+    service: Service = wrapper.session.query(Service).filter_by(uuid=data["service_uuid"]).first()
+
+    given_per: Tuple[float, float, float, float, float] = game_content.dict2tuple(data)
+
+    expected_per: Tuple[float, float, float, float, float] = game_content.dict2tuple(
+        config["services"][service.name]["needs"]
+    )
+
+    speed: float = game_content.calculate_speed(expected_per, given_per)
+
+    service.speed = (service.speed + speed) / 2
```

Comment:
why not set the speed directly?
```suggestion
    service.speed = speed
```

Reply:
The reason why I wrote it that way is that it is a more accurate way of "describing" the process because speed describes how fast a process runs and I would be abuse able too set it directly. Run many Processes -> Gather Progress Time with low speed ... kill Some of them -> get high speed finish process way better results.